src/main/java/gomedic/logic/parser/findcommandparser/FindPatientCommandParser.java

```diff
@@ -0,0 +1,78 @@
+package gomedic.logic.parser.findcommandparser;
+
+import gomedic.commons.core.Messages;
+import gomedic.logic.commands.findcommand.FindPatientCommand;
+import gomedic.logic.parser.Parser;
+import gomedic.logic.parser.exceptions.ParseException;
+import gomedic.model.util.*;
+
+
+import java.util.Arrays;
+
+import static gomedic.logic.parser.CliSyntax.*;
+
+/**
+ * Parses input arguments and creates a new FindPatientCommand object
+ */
+public class FindPatientCommandParser implements Parser<FindPatientCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindPatientCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindPatientCommand.MESSAGE_USAGE));
+        }
+
+        // First find out which field it is supposed to match to
+        // then create a PatientCommand with the argument containing the corresponding
+        // Predicate
+        if (trimmedArgs.indexOf("/") == -1 ) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindPatientCommand.MESSAGE_USAGE));
+        }
+        String[] fieldAndArguments = trimmedArgs.split("/");
+        if (fieldAndArguments.length != 2) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindPatientCommand.MESSAGE_USAGE));
+        }
+        fieldAndArguments[1] = fieldAndArguments[1].trim();
+        String[] argumentKeywords = fieldAndArguments[1].split("\\s+");
+
+        String field = fieldAndArguments[0];
+
+        switch (field) {
```

Comment:
What happens if I have `find t/patient n/Simon age/50`, wouldn't this just return the find t/patient by name ?

Reply:
The find function currently supports searching via 1 field only